spec.html

```diff
@@ -20653,20 +21261,269 @@ <h1>Static Semantics: PrototypePropertyNameList</h1>
+      <emu-alg>
+        1. For each child node _child_ of this Parse Node, do
+          1. If _child_ is an instance of a nonterminal, then
+            1. If AllPrivateIdentifiersValid of _child_ with argument _names_ is *false*, return *false*.
+        1. Return *true*.
+      </emu-alg>
+
+      <emu-grammar>MemberExpression : MemberExpression `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
+          1. Return AllPrivateIdentifiersValid of |MemberExpression| with argument _names_.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>CallExpression : CallExpression `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
+          1. Return AllPrivateIdentifiersValid of |CallExpression| with argument _names_.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>OptionalChain : `?.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, return *true*.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>OptionalChain : OptionalChain `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
+          1. Return AllPrivateIdentifiersValid of |OptionalChain| with argument _names_.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>ClassBody : ClassElementList</emu-grammar>
+      <emu-alg>
+        1. Let _newNames_ be a copy of _names_.
+        1. Append to _newNames_ the elements of PrivateBoundIdentifiers of |ClassBody|.
+        1. Return AllPrivateIdentifiersValid of |ClassElementList| with argument _newNames_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-class-definitions-static-semantics-privateboundidentifiers" type="sdo" aoid="PrivateBoundIdentifiers">
+      <h1>Static Semantics: PrivateBoundIdentifiers</h1>
+      <emu-grammar>
+        FieldDefinition : ClassElementName Initializer?
+      </emu-grammar>
+      <emu-alg>
+        1. Return PrivateBoundIdentifiers of |ClassElementName|.
+      </emu-alg>
+
+      <emu-grammar>
+        ClassElementName : PrivateIdentifier
+      </emu-grammar>
+      <emu-alg>
+        1. Return a List whose sole element is the StringValue of |PrivateIdentifier|.
+      </emu-alg>
+
+      <emu-grammar>
+        ClassElementName : PropertyName
+
+        ClassElement : `;`
+      </emu-grammar>
+      <emu-alg>
+        1. Return a new empty List.
+      </emu-alg>
+
+      <emu-grammar>
+        ClassElementList : ClassElementList ClassElement
+      </emu-grammar>
+      <emu-alg>
+        1. Let _names_ be PrivateBoundIdentifiers of |ClassElementList|.
+        1. Append to _names_ the elements of PrivateBoundIdentifiers of |ClassElement|.
```

Comment:
is this mutating? or should it be making a copy of _names_?

Reply:
There is no observable difference. This pattern is what we do for similar cases, like [BoundNames](https://tc39.es/ecma262/#sec-static-semantics-boundnames).